src/snowflake/cli/plugins/object/stage/commands.py

```diff
@@ -71,20 +77,34 @@ def copy(
 
     if is_get:
         target = Path(destination_path).resolve()
-        cursor = StageManager().get(
-            stage_path=source_path, dest_path=target, parallel=parallel
-        )
-    else:
-        source = Path(source_path).resolve()
-        local_path = str(source) + "/*" if source.is_dir() else str(source)
-
-        cursor = StageManager().put(
-            local_path=local_path,
-            stage_path=destination_path,
-            overwrite=overwrite,
-            parallel=parallel,
-        )
-    return QueryResult(cursor)
+        if recursive:
+            cursors = StageManager().get_recursive(
+                stage_path=source_path, dest_path=target, parallel=parallel
+            )
+            return MultipleResults([QueryResult(c) for c in cursors])
+        else:
+            cli_console.warning(
+                "Use `--recursive` flag, which copy files recursively with directory structure. This will be the default behavior in the future."
+            )
+            cursor = StageManager().get(
+                stage_path=source_path, dest_path=target, parallel=parallel
+            )
+            return QueryResult(cursor)
+
+    if is_put:
+        if recursive:
+            raise click.ClickException("Recursive flag for upload is not supported.")
+        else:
+            source = Path(source_path).resolve()
+            local_path = str(source) + "/*" if source.is_dir() else str(source)
+
+            cursor = StageManager().put(
+                local_path=local_path,
+                stage_path=destination_path,
+                overwrite=overwrite,
+                parallel=parallel,
+            )
+            return QueryResult(cursor)
```

Comment:
Bad design, if for some reason we do not enter any of the if the function will return None causing unexpected behavior